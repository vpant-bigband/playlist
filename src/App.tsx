import React, { useState, useCallback } from 'react';
import { Header } from './components/Header';
import VideoCardList from './components/VideoCardList/VideoCardList';
import Loader from './components/Loader/Loader';
import useApp from './useApp.hooks';
import { ErrorBoundary } from './components/ErrorBoundary';

const App: React.FunctionComponent = () => {
  const [keyword, setKeyword] = useState<string>('');
  const onSearchIconClick = useCallback((searchText) => setKeyword(searchText), []);  
  const [videoCardListData, isLoading, onScrollEndHandler, errorInfo] = useApp(keyword);
  // console.log('videoCardListData ', videoCardListData);
  return (
    <>
      <Header onSearchIconClick={onSearchIconClick}/>
      <main>
        <ErrorBoundary>
          { !errorInfo &&
            (
            <>
              <VideoCardList onScrollEndHandler={onScrollEndHandler} data={videoCardListData} />
              {isLoading && <Loader />}
            </>
            )
          } 
          {
            errorInfo && errorInfo.message
          }
        </ErrorBoundary>
      </main>
    </>
  );
}

export default App;
