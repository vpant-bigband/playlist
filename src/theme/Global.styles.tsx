import React from 'react';
import { Global, css } from '@emotion/react';
import theme from './theme.constant';

const GlobalStyle: React.FunctionComponent = () => (
  <Global
    styles={css`
      body {
        font-family: ${theme.fonts.sansSerif};
        font-size: ${theme.fontSizes[2]}px;
        line-height: ${theme.lineHeights.title};
        margin: 0px;
      }
      main {
        max-width: 1440px;
        margin-left: auto;
        margin-right: auto;
      }
    `}
  />
);

export default GlobalStyle;
