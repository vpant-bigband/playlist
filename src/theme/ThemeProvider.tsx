import React from 'react';
import { ThemeProvider } from '@emotion/react';
import ThemeConstant, { ThemeType } from './theme.constant';
import GlobalStyle from './Global.styles';

interface ThemeProps {
  children: React.ReactChildren | React.ReactElement | React.ReactElement[];
};

const Theme: React.FunctionComponent<ThemeProps> = ({ children }) => {
  return (
    <ThemeProvider theme={ThemeConstant as ThemeType}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
}



export default Theme;