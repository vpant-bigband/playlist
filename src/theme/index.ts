export { default as Theme } from './ThemeProvider';