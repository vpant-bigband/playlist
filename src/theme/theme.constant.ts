const theme = {
  breakpoints: ['576px', '767px', '992px', '1200px'],
  BREAKPOINTS: [576, 767, 992, 1200],
  space: [0, 4, 8, 16, 24, 32, 64, 128, 256, 512],
  fontSizes: [10, 12, 14, 16, 18, 20, 24, 36, 48, 80, 96],
  fontWeights: [100, 200, 300, 400, 500, 600, 700, 800, 900],
  lineHeights: {
    solid: 1,
    title: 1.25,
    copy: 1.5,
  },
  fonts: {
    serif: 'athelas, georgia, times, serif',
    sansSerif: '"Roboto", sans-serif',
    montserrat: 'Montserrat',
  },
  borders: [0, '1px solid', '2px solid', '4px solid', '8px solid', '16px solid', '32px solid'],
  width: [16, 24, 32, 64, 128, 256],
  heights: [16, 24, 32, 64, 128, 256],
  maxWidths: [16, 32, 64, 128, 256, 512, 768, 1024, 1536],
  colors: {
    // Copy Typography
    blue: '#0E5AA7',
    darkBlue: '#0a437c',
    purple: '#6F1C60',
    lightBlue: '#b7cee5',
    deepRed: '#8A0C0C',
    // Greys
    ashGray: '#F9F7F7',
    lightGray: '#EAEAEA',
    darkGray: '#2D2D2D',

    white: '#fff',
    transparent: 'transparent',
    black: '#000',
    // Notification color
    error: '#FEF4F4',
    warning: '#FCF6E7',
    information: '#ECF3FA',
    success: '#F1FAF9',
  },
};

export type ThemeType = typeof theme;

export default theme;