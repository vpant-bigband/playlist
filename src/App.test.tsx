import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Theme } from './theme';
import SuccessResponseMock from './__mocks__/app/SuccessResponse.mock';
import VideoCardListMock from './components/VideoCardList/__mocks__/VideoCardList.mock';
import { intersectionMockInstance } from 'react-intersection-observer/test-utils';
import { renderHook } from '@testing-library/react-hooks';
import useApp from './useApp.hooks';


const mockIntersectionObserver = jest.fn();

mockIntersectionObserver.mockReturnValue({
  observe: intersectionMockInstance,
  unobserve: intersectionMockInstance,
  disconnect: intersectionMockInstance,
});

window.IntersectionObserver = mockIntersectionObserver;

// @ts-ignore
window.isIntesecting = true
const renderSetup = () => render(<Theme><App /></Theme>);

const onScrollEndHandler = jest.fn();

// jest.mock('./useApp.hooks', () => {
//   return jest.fn(() => [SuccessResponseMock, false, onScrollEndHandler]);
// });
describe('App test suite', () => {

  beforeEach(() => {
     // @ts-ignore
     fetch.mockResponse(JSON.stringify(VideoCardListMock));
  });
  it('should match snapshot', () => {
    const { container } = renderSetup();
    expect(container).toMatchSnapshot();
  });

  it('should ', async () => {
    const { result, waitForValueToChange } = renderHook(() => useApp('music'));
    await waitForValueToChange(() => {
      return result.current[0];
    })
    const { debug } = renderSetup();
    console.log(debug());
  });
});
