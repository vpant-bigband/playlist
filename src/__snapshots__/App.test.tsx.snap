// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite should match snapshot 1`] = `
<div>
  <div
    class="css-nd3dh8"
  >
    <div
      class="css-yn2c7i"
      font-size="1,2,3,4"
    >
      <input
        class="css-19n93a3"
        placeholder="Search for videos"
      />
      <span
        class="css-ntsb59"
        width="20%,,10%,12%,8%"
      >
        <svg
          data-testid="magnify-glass-icon"
          height="40"
          viewBox="0 0 16 16"
          width="40"
        >
          <path
            d="M14.23437,15.10449l-3.2539-3.25488a6.13908,6.13908,0,1,1,1.63574-1.85352c-.05859.10156-.11816.19922-.17969.29492l-.63086-.40625c.05567-.08593.10938-.17383.16016-.26269a5.406,5.406,0,1,0-1.19922,1.43945.37776.37776,0,0,1,.50781.02051l3.49121,3.49121Z"
            fill="#fff"
          />
        </svg>
      </span>
    </div>
  </div>
  <main />
</div>
`;
